ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/SPOTResponseHandler.java

```diff
@@ -33,29 +35,30 @@ public SPOTResponseHandler(DynamoSpotService dynamoSpotService) {
     }
 
     @Override
-    public Object handleRequest(SNSEvent input, Context context) {
-        input.getRecords().stream()
-                .map(SNSEvent.SNSRecord::getSNS)
-                .map(SNSEvent.SNS::getMessage)
-                .map(this::processSPOTResponse)
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .forEach(this::writeToDynamo);
-
-        return null;
-    }
-
-    private Optional<SPOTResponse> processSPOTResponse(String message) {
-        try {
-            return Optional.of(objectMapper.readValue(message, SPOTResponse.class));
-        } catch (JsonProcessingException e) {
-            LOG.error("Unable to deserialize SPOT response");
-            return Optional.empty();
+    public Object handleRequest(SQSEvent event, Context context) {
+        for (SQSMessage msg : event.getRecords()) {
+            try {
+                var spotResponse = objectMapper.readValue(msg.getBody(), SPOTResponse.class);
+                if (!spotResponse.getStatus().equals(SPOTStatus.OK.toString())) {
```

Comment:
Rather than comparing the string values, can we deserialise the status directly into a `SPOTStatus1?

Then we can do `if (spotResponse.getStatus() != SPOTStatus.OK) {...}`

Reply:
Thanks @mrwilson. I have amended.